syntax = "proto3";

import "alice/boltalka/generative/service/proto/common.proto";

package NGenerativeBoltalka.Proto;

message TGenerativeRequestSeed {
    uint32 Value = 1;
}

message TSamplerParams {
    optional string Mode = 1;
    optional float Temperature = 2;
    optional uint64 TopK = 3;
    optional float Nucleus = 4;
};

message TGenerativeRequest {
    repeated string Context = 1;
    int32 NumHypos = 2;
    TGenerativeRequestSeed Seed = 3;
    bool PrefixTokensOnly = 4;
    bool DiverseBeamSearch = 5;
    repeated string SpanDelimiters = 6;
    TPtuneEmbeddings PtuneEmbeddings = 7;
    TSamplerParams SamplerParams = 8;
    optional uint32 MinOutLen = 9;
    optional uint32 MaxOutLen = 10;
}
